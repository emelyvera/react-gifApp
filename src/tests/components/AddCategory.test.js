import React from 'react';
import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
;



describe('Pruebas en el componente <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should de mostrarse coreectamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('should de cambiar la caja de texto handleInputChange', () => {

        const input = wrapper.find('input');
        const value = 'HolaMundo'
        input.simulate('change', { target: { value } } );

        expect( wrapper.find('p').text().trim() ).toBe(value);
    });

    test('NO should de postear la informacion con submit handleSubmit', () => {
      
      wrapper.find('form').simulate('submit', { preventDefault(){} })  ;
      expect( setCategories ).not.toHaveBeenCalled();

    });

    test('should llamar la funcion setcategories y limpiar la caj de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'HolaMundo'
        input.simulate('change', { target: { value } } ); 

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( input.prop('value').trim() ).toBe('');
        
    });
    
    


});
