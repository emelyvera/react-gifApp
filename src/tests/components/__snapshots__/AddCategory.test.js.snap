// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> should de mostrarse coreectamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
